src/Bridge/Elasticsearch/DataProvider/Filter/TermFilter.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Elasticsearch\DataProvider\Filter;
+
+use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Filter the collection by given properties.
+ *
+ * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-term-query.html
+ *
+ * @experimental
+ *
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ */
+final class TermFilter extends AbstractFilter implements ConstantScoreFilterInterface
+{
+    private $identifiersExtractor;
+    private $iriConverter;
+    private $propertyAccessor;
+
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceClassResolverInterface $resourceClassResolver, IdentifiersExtractorInterface $identifiersExtractor, IriConverterInterface $iriConverter, PropertyAccessorInterface $propertyAccessor, array $properties = null)
+    {
+        parent::__construct($propertyNameCollectionFactory, $propertyMetadataFactory, $resourceClassResolver, $properties);
+
+        $this->identifiersExtractor = $identifiersExtractor;
+        $this->iriConverter = $iriConverter;
+        $this->propertyAccessor = $propertyAccessor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(array &$clauseBody, string $resourceClass, string $operationName = null, array $context): void
+    {
+        $terms = [];
+
+        foreach ($context['filters'] ?? [] as $property => $values) {
```

Comment:
This is not consistent with the "doctrine" filters.
The context of a doctrine-filter service are declared like `['foo' => 'exact', 'bar' => '~']` but this implementation supports the following syntax `['foo', 'bar']`.